### PR TITLE
    MGMT-18442: Add service account authentication mechanism alongside offline token

### PR DESCRIPTION
**Pull request by danmanor**
This PR is mainly about adding service account authentication mechanism alongside offline tokens as offline tokens will soon be deprecated. In addition, this PR fixed minor related issues:

1. Usage of old OCM UI endpoint (`cloud...`)
2. Add the required changes in test-infra for - https://github.com/openshift/assisted-service/pull/6307
3. Docs

### CODE REVIEW

File: skipper.env [hunk 1]
```diff
@@ -189,3 +189,6 @@ HOST_INSTALLER_ARGS
 ENABLE_SKIP_MCO_REBOOT
 ENABLE_SOFT_TIMEOUTS
 IMAGES_FLAVOR
+SERVICE_ACCOUNT_CLIENT_ID
```
Comment (adriengentil) [inline skipper.env:192]: remove `OFFLINE_TOKEN` ? I see we also have `SECOND_OFFLINE_TOKEN` :thinking: maybe it's worth to remove it as well?

File: src/tests/global_variables/env_variables_defaults.py [hunk 2]
```diff
@@ -18,7 +18,8 @@ class _EnvVariables(DataPool, ABC):
     remote_service_url: EnvVar = EnvVar(["REMOTE_SERVICE_URL"])
     pull_secret: EnvVar = EnvVar(["PULL_SECRET"])
     deploy_target: EnvVar = EnvVar(["DEPLOY_TARGET"])
-    offline_token: EnvVar = EnvVar(["OFFLINE_TOKEN"])
```
Comment (adriengentil) [inline src/tests/global_variables/env_variables_defaults.py:21]: `OCM_SELF_TOKEN` isn't used?

**Comment by danmanor** [comment]
/override ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node

File: src/service_client/assisted_service_api.py [hunk 3]
```diff
@@ -13,19 +14,47 @@
...
 
+class ServiceAccount(BaseModel):
+    client_id: Optional[str]
+    client_secret: Optional[str]
+
+    def __hash__(self):
+        return hash((self.client_id, self.client_secret))
+
+    def __eq__(self, other):
+        if isinstance(other, ServiceAccount):
```
Comment (eifrach) [inline src/service_client/assisted_service_api.py:32]:
```suggestion
        if isinstance(other, ServiceAccount) -> bool:
```

File: src/service_client/assisted_service_api.py [hunk 4]
```diff
@@ -47,10 +76,16 @@ def get_host(self, configs: Configuration) -> str:
...
     @classmethod
-    def set_config_auth(cls, c: Configuration, offline_token: Optional[str]) -> None:
-        if not offline_token:
-            log.info("OFFLINE_TOKEN not set, skipping authentication headers")
-            return
+    def set_config_auth(
+        cls, c: Configuration, offline_token: Optional[str], service_account: Optional[ServiceAccount]
+    ) -> None:
+        authentication_method = AuthenticationMethod.SERVICE_ACCOUNT
+        if service_account is None or not service_account.is_provided():
```
Comment (eifrach) [inline src/service_client/assisted_service_api.py:83]: This is a bit confusing, setting the method as SERIVCE_ACCOUNT before validating the method type

**Comment by danmanor** [comment]
/override ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node

**Pull request merged**